### ISSUE TITLE
Official vector graphics

### ISSUE THREAD

**Issue by SuibianP** [joulev/mrt-badges#1]
Thanks for the work! I am trying to do some SG MRT 音MAD and this helps a lot!

FWIW, the official vector graphics are actually available as various assets of <https://www.lta.gov.sg/content/ltagov/en/map/train.html>, including the [full system map](https://www.lta.gov.sg/map/mrt/Full%20map-color.svg) and the badges of individual current lines e.g. <https://www.lta.gov.sg/content/dam/ltagov/img/map/mrt/Icon_Sengkang_LRT.svg>. The fonts are vectorised though.

The official vector graphics of upcoming lines might have been available at the [LTA Connecting People, Places & Possibilities exhibition](https://www.lta.gov.sg/content/ltagov/en/who_we_are/our_work/connecting_people_places_possibilities_exhibition.html) which included [an interactive future MRT map](https://landtransportguru.net/future-cross-island-line-stations-revealed-in-lta-exhibition/), but I cannot find any functional archives of <https://www.connectingppp.lta.gov.sg> thus unable to verify.

**Comment by joulev** [comment]
Yes, actually I also managed to get the full system map in SVG format earlier shortly after Teck Lee LRT opened: [Download (1.2 MB)](https://r2.joulev.dev/files/mcazpbyuhz2wr0kwi8yqmkms). LTA used to host this SVG file directly but apparently shifted to [a PDF](https://www.lta.gov.sg/content/dam/ltagov/getting_around/public_transport/rail_network/pdf/SM_TEL4_Eng_(Ver150824)_TL.pdf) now.

Annoyingly, LTA doesn't seem to use one single shade of colour throughout their maps. For the North East Line, in the SVG above the hex is #9900AB, but in a PDF map from TEL3, the North East Line is #8B2D9A. And in [another vector file](https://www.lta.gov.sg/content/dam/ltagov/img/map/mrt/Icon_NorthEastLine.svg), the hex is #9E28B5. Extremely similar colours, [the difference can hardly be seen with the naked eye](https://coolors.co/9e28b5-9900ab-8b2d9a), but they are still different colours. And when we don't have a single colour used officially everywhere, we can't call any of these colours *the* official colour.

Also none of the official vector maps have the colours of JRL (teal) and CRL (lime) under construction. For CRL, I at least have [this raster image](https://r2.joulev.dev/files/xhli8sjarotbo1rcb6lhx4er) with a reasonable amount of evenness at the bottom to use an eyedropper on, so the shade of CRL used in this repo should be close enough. For JRL I can't for the life of me find a high-quality image where eyedropper doesn't change colour every pixel, so... yeah. Jurong Region Line colour shade is still unknown.

**Comment by SuibianP** [comment]
JRL seems to be #00B0BE based on the [Location of Stations](https://www.lta.gov.sg/content/ltagov/en/upcoming_projects/rail_expansion/jurong_region_line.html#collapse-content-tab_yellow_new_904966419-LocationofStations) illustrations, which is relatively uniform.

I have submitted an enquiry to LTA, not sure if they would care to clarify though...

**Comment by joulev** [comment]
Unfortunately not uniform enough... I get these just from the region around "JR1" in Choa Chu Kang

<img width="133" alt="image" src="https://github.com/user-attachments/assets/bb47fa2f-429d-4cc1-b394-d9f416f3812c">

Hopefully LTA will reply. Previously they replied to the public asking whether CCL is orange or yellow, so there is hope.

It would be best if they just released an updated map like CRL, but I guess that requires the station names to be finalised, and the polls for the station names have not even been conducted yet.

**Comment by SuibianP** [comment]
I meant the illustrations of individual stations, such as ![](https://www.lta.gov.sg/content/dam/ltagov/upcoming_projects/rail_expansion/JRL/LTA_MRT-JS1_ChoaChuKang_270819.png)

Those have a decent level of uniformity at the station fill and to the bottom right corner.

**Comment by joulev** [comment]
OHH yeah you are right! Updated the source code, thanks!

That said, it's still not perfectly uniform (which is not surprising for LTA, to be honest). I got #00b0be and #00b0bd from two different spots. Though since it is only different by one single point over 2^24, so it is as close as we can get already, considering LTA themselves don't use the same shade everywhere.

**Comment by joulev** [comment]
Update: I found [this document](https://jooferj.github.io/resources/tsm/LTA%20Transit%20Signage%20Manual%20Volume%201.pdf) in the wild, and it seems very likely to be official. Among other things it lists the official hex codes of all line colours (along with other colours used in the TEL-era MRT signage system), which means we can now, finally, have a (99% confirmed) source guideline to base the colours on.

Line colours in RGB syntax:

|      | Background              | Foreground (presumed) |
|------|-------------------------|-----------------------|
| NSL  | 223 40 39 (`#df2827`)   | white                 |
| EWL  | 0 150 69 (`#009645`)    | white                 |
| NEL  | 153 0 171 (`#9900ab`)   | white                 |
| CCL  | 250 158 13 (`#fa9e0d`)  | 46 42 37 (`#2e2a25`)  |
| DTL  | 0 85 184 (`#0055b8`)    | white                 |
| TEL  | 157 89 24 (`#9d5918`)   | white                 |
| JRL  | 0 173 187 (`#00adbb`)   | white                 |
| CRL  | 147 213 0 (`#93d500`)   | 46 42 37 (`#2e2a25`)  |
| LRT  | 112 130 112 (`#708270`) | white                 |
| Bus  | 147 213 0 (`#93d500`)   | 46 42 37 (`#2e2a25`)  |
| Exit | 255 209 0 (`#ffd100`)   | 46 42 37 (`#2e2a25`)  |
